sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/EventHubsTemplate.java

```diff
@@ -54,19 +57,39 @@ public EventHubsTemplate(EventHubsProducerFactory producerFactory) {
         EventHubProducerAsyncClient producer = producerFactory.createProducer(destination);
         CreateBatchOptions options = buildCreateBatchOptions(partitionSupplier);
 
-        return producer.createBatch(options).flatMap(batch -> {
-            for (EventData event : events) {
-                try {
-                    batch.tryAdd(event);
-                } catch (AmqpException e) {
-                    LOGGER.error("Event is larger than maximum allowed size. Exception: " + e);
-                }
-            }
-            return producer.send(batch);
-        });
+        AtomicReference<EventDataBatch> currentBatch = new AtomicReference<>(
+                producer.createBatch(options).block());
+
+        Flux.fromIterable(events).flatMap(event -> {
+                    final EventDataBatch batch = currentBatch.get();
+                    if (batch.tryAdd(event)) {
+                        return Mono.empty();
```

Comment:
This is not correct, if we return `Mono.empty()`, the subscriber of this mono won't be able to perform actions upon the signal.

Reply:
concerning the the nature of this method, I fix the return statement in another place. seems return Mono which holds the sending response void for the last batch for the seq of events.  In this place, Mono.empty() is in a Flux , this is no concern for if it is able to perform actions upon the signal.

in the return, the sending method return Mono just like orignal.

thing to notice, the previous batch will be executed sync before the last sending async. This is the current .